libcomposefs/lcfs-utils.h

```diff
@@ -19,12 +19,41 @@
 
 #include <assert.h>
 #include <stdlib.h>
+#include <string.h>
 #include <errno.h>
 #include <unistd.h>
 
 #define max(a, b) ((a > b) ? (a) : (b))
 #define min(a, b) (((a) < (b)) ? (a) : (b))
 
+static inline bool str_has_prefix(const char *str, const char *prefix)
+{
+	return strncmp(str, prefix, strlen(prefix)) == 0;
+}
+
+static inline char *memdup(const char *s, size_t len)
+{
+	char *s2 = malloc(len);
+	if (s2 == NULL) {
+		errno = ENOMEM;
+		return NULL;
+	}
+	memcpy(s2, s, len);
+	return s2;
+}
+
+static inline char *str_join(const char *a, const char *b)
```

Comment:
optional: This is just code motion, so OK, but same question re `asprintf` - can be a followup.